### DISCUSSION TITLE
Distinguish between marking failure/success by user and by task

### DISCUSSION THREAD

**Discussion by guidopetri** [apache/airflow#33581] [Ideas]
### Description

Tasks should have a marker on them or additional information showing that they were marked success/failure/run/cleared by hand vs by the scheduler/worker processes.

### Use case/motivation

When working with others on Airflow DAGs, we constantly have to keep in communication as to what we marked succeeded/failed because otherwise it is not clear whether a task succeeded/failed "naturally" as per the task or "artificially" as being marked as such by a user.

It would be nice to have a difference between the two states, or a marker of some sort to identify that a task was succeeded/failed by hand.

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by potiuk** [comment]
Sounds like an interesting feature. Maybe you would like to add it @guidopetri ?

**Comment by guidopetri** [comment]
I'm not sure I have the required knowledge for that.

There's an enum of task states [here](https://github.com/apache/airflow/blob/1d170f899bcc87110e55192517270ec89d511ca8/airflow/utils/state.py#L26), which we could add to, but that would imply changing the marking mechanism to e.g. `MARKED_SUCCESS` or `MARKED_FAILED` instead of `SUCCESS` or `FAILED`. Which would then break anything that depends on the task instances being one of the latter two states. That sounds like a lot bigger of a change than I am ready for...

**Comment by potiuk** [comment]
> I'm not sure I have the required knowledge for that.

Understood - maybe someone will pick it up then if there is an interest.

**Comment by kurtqq** [comment]
> There's an enum of task states [here](https://github.com/apache/airflow/blob/1d170f899bcc87110e55192517270ec89d511ca8/airflow/utils/state.py#L26), which we could add to, but that would imply changing the marking mechanism to e.g. `MARKED_SUCCESS` or `MARKED_FAILED` instead of `SUCCESS` or `FAILED`. Which would then break anything that depends on the task instances being one of the latter two states. That sounds like a lot bigger of a change than I am ready for...

The concept of adding explicit status per each possible action is dangerous. When start this you will never hear the end of it.
To answer of the question who did something should be stored in logs. If needed it should be displayed from the logs.
When you know who did it you also get the answer about what he did because if a user marked task as failed clearly this is done manually so you get this way more information without adding more statuses.

**Comment by utkarsharma2** [comment]
@potiuk , I would like to work on this, can you please assign this ticket to me?
